es6/lib/createRouteHandler.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+// jshint -W040
+
+import Boom from 'boom';
+import first from 'lodash/array/first';
+import forEach from 'lodash/collection/forEach';
+import get from 'lodash/object/get';
+import isEmpty from 'lodash/lang/isEmpty';
+import isUndefined from 'lodash/lang/isUndefined';
+import omit from 'lodash/object/omit';
+
+function apiController(
+  knexClient,
+  tableName,
+  tableIndex,
+  requestTransformFunction = value => value,
+  responseTransformFunction = value => value
+) {
+  return {
+
+    create(request, reply) {
+      let payload = get(request.pre, 'customRequest');
+
+      if(isUndefined(payload)) {
+        payload = request.payload;
+      }
+
+      function createApiResponse(response) {
+        return responseTransformFunction({ [tableIndex]: first(response) });
+      }
+
+      return knexClient
+        .table(tableName)
+        .insert(requestTransformFunction(payload))
+        .then(createApiResponse)
+        .then(reply)
+        .catch(reply);
+    },
+
+    destroy(request, reply) {
+      function createDeleteResponse(result) {
+        if (result === 1 || result === 0) {
+          reply().code(204);
+          return;
+        }
+        reply().code(500);
+      }
+
+      return knexClient
+        .table(tableName)
+        .where(tableIndex, request.params.id)
+        .del()
+        .then(createDeleteResponse)
+        .catch(reply);
+    },
+
+    list(request, reply) {
+      let query = get(request.pre, 'customRequest');
+
+      if(isUndefined(query)) {
+        query = request.query;
+      }
+
+      if(isUndefined(get(query, 'limit'))) {
+        query.limit = 500;
+      }
+
+      if(isUndefined(get(query, 'cursor'))) {
+        query.cursor = 1;
+      }
+
+      query = requestTransformFunction(query);
+
+      function filterQuery() {
+        function constructWhere(value, key) {
+          this.where(key, value);
+        }
+
+        forEach(
+          omit(query, ['cursor', 'limit']),
+          constructWhere,
+          this
+        );
+      }
+
+      function prepareResponse(result) {
+        return responseTransformFunction(result);
+      }
+
+      function createApiResponse(result) {
+        let apiResponse = {
+          limit: null,
+          cursor: null,
+          records: result
+        };
+
+        if (isEmpty(result)) {
+          reply(apiResponse);
+          return;
+        }
+
+        apiResponse.limit = query.limit;
+
+        if (result.length === query.limit) {
+          apiResponse.cursor = query.cursor + 1;
+        }
+
+        reply(apiResponse);
+      }
+
+      return knexClient
+        .table(tableName)
+        .where(filterQuery)
+        .limit(query.limit)
+        .offset((query.cursor - 1) * query.limit)
+        .map(prepareResponse)
+        .then(createApiResponse)
```

Comment:
Handle like you did in create where it returns the response and doesn't do `reply` for you.